### DISCUSSION TITLE
How to intercept https://archive.is with mitmproxy?

### DISCUSSION THREAD

**Discussion by rain1** [mitmproxy/mitmproxy#6570] [Q&A]
Hi!

When I visit https://archive.is/ without proxy then it loads fine.

When I visit it with mitmdump then it loads very long time until finally my browser says 502 Bad Gateway and mitmdump says
```
[01:54:52.624][[::1]:52326] client connect
[01:54:52.677][[::1]:52326] server connect archive.is:443 (5.42.83.165:443)
[::1]:52326: GET https://archive.is/ HTTP/2.0
 << peer closed connection
[01:55:52.723][[::1]:52326] server disconnect archive.is:443 (5.42.83.165:443)
[01:58:43.370][[::1]:52326] client disconnect
```

How to fix that?

**Comment by Prinzhorn** [comment]
I've opened https://github.com/mitmproxy/mitmproxy/issues/6571 as this appears to be a bug in HTTP/2 implementation. Either on our end or with https://archive.is/

**Reply by rain1** [reply]
Thanks for creating a ticket.  I will remember to check it every time I run into this site or till it gets fixed.....

Isn't there any way to bypass this HTTP/2 thing?

I usually append those command line parameters:
```
    sys.argv.append("--ssl-insecure")
    sys.argv.append("--no-http2")
    sys.argv.append("--set")
    sys.argv.append("connection_strategy=lazy")
```
But commenting them out and restarting didn't help. I was just curious that since you mentioned HTTP/2 that this --no-http2 did not seem to have effect?

**Reply by Prinzhorn** [reply]
You're right, I misinterpreted the results. You can't fetch the site with curl at all, even without mitmproxy. I've updated the issue https://github.com/mitmproxy/mitmproxy/issues/6571#issuecomment-1874907934 . Feel free to reply there if you find out why that is.
